### PR TITLE
Improved OptimizationResult

### PR DESCRIPTION
**Pull request by regislebrun**
Now it gives access to the associated optimization problem, useful to check which constraint is active or for graphical post-processing.

### CODE REVIEW

File: lib/src/Base/Optim/openturns/OptimizationResult.hxx [hunk 1]
```diff
@@ -47,12 +48,12 @@ public:
...
-      const NumericalScalar absoluteError,
-      const NumericalScalar relativeError,
-      const NumericalScalar residualError,
-      const NumericalScalar constraintError);
+                     const NumericalPoint & optimalValue,
+                     const UnsignedInteger iterationNumber,
+                     const NumericalScalar absoluteError,
+                     const NumericalScalar relativeError,
+                     const NumericalScalar residualError,
+                     const NumericalScalar constraintError, const OptimizationProblem & problem);
```
Comment (jschueller) [inline lib/src/Base/Optim/openturns/OptimizationResult.hxx]: align ?

**Comment by jschueller** [comment]
+1

**Comment by jschueller** [comment]
@regislebrun you could make sure that your email address that you use for commits (git config user.email) is associated to your github account for commit stats to be taken into account.

File: lib/src/Base/Optim/OptimizationResult.cxx [hunk 2]
```diff
@@ -58,7 +59,8 @@ OptimizationResult::OptimizationResult(const NumericalPoint & optimalPoint,
     const NumericalScalar absoluteError,
     const NumericalScalar relativeError,
     const NumericalScalar residualError,
-    const NumericalScalar constraintError)
+				       const NumericalScalar constraintError,
+const OptimizationProblem & problem)
```
Comment (dbarbier) [inline lib/src/Base/Optim/OptimizationResult.cxx]: Indentation

**Comment by dbarbier** [comment]
+1

**Comment by sofianehaddad** [comment]
+1

**Pull request merged**